### DISCUSSION TITLE
Format SD card as a command without user confirmation

### DISCUSSION THREAD

**Discussion by postbreak** [gopro/labs#264] [Feature Requests]
Hello would it be possible to add a format SD card, or delete last recorded video file as a command without user interaction?

When using a GoPro as a dashcam, with power on with USB, and looping it works 99%. The 1% that's missing is SD card looping, so after X amount of boots and recording a loop, the SD card fills with no way to automatically clear.

I wish GoPro would embrace the dashcam use as it wipes the floor with every dash cam out there. I know it's not "on brand" but for a nerd like me, "!FORMATQUIET" or something like that I could run at boot, or after X amount of boots, it would make it viable as a real dash cam.

**Comment by dnewman-gpsw** [comment]
This is to prevent bad actors creating malicious QR Codes that delete media. But the problem you describe is a valid one, it would be nice to have a way do support recording new media, even if the camera is full (or near full.)  

For Labs, it could be a boot command that deletes the oldest files up say 64GB, or half the card, which ever is smaller.  This way, malicious  actions are limited, and the dashcam use case is solved.  I will add this to the ideas backlog.

**Reply by postbreak** [reply]
Thank you, I will try to brainstorm some other non-destructive ideas that would allow this to function. Maybe you can make a subfolder on the card that is either a symbolic link that is temporary or files inside the subfolder are able to be overwritten without having to worry about formatting the whole card. 

Deleting the oldest files on boot would be pretty much perfect though. With looping set at 20 minutes, and a 128GB card, you'd be able to have a nice chunk of previous data, and the oldest file being deleted would free the card of enough space to continue recording. There could be a race condition if the oldest file is smaller than the new recording so it would take some testing to see how it works in practice.

Thank you for considering this feature, I know so many of my friends who want to use their GoPro as a set and forget dash cam and this one feature is the missing link. It would be huge to get it working.
